jerry-core/ecma/builtin-objects/ecma-builtin-regexp-prototype.cpp

```diff
@@ -45,6 +45,173 @@
+  {
+    /* Compile can only be called on RegExp objects. */
+    ret_value = ecma_make_throw_obj_completion_value (ecma_new_standard_error (ECMA_ERROR_TYPE));
+  }
+  else
+  {
+    ecma_string_t *pattern_string_p = NULL;
+    uint8_t flags = 0;
+
+    if (ecma_is_value_object (pattern_arg)
+        && ecma_object_get_class_name (ecma_get_object_from_value (pattern_arg)) == LIT_MAGIC_STRING_REGEXP_UL)
+    {
+      if (!ecma_is_value_undefined (flags_arg))
+      {
+        ret_value = ecma_raise_type_error ("Invalid argument of RegExp compile.");
+      }
+      else
+      {
+        /* Compile from existing RegExp pbject. */
+        ecma_object_t *target_p = ecma_get_object_from_value (pattern_arg);
+
+        /* Get source. */
+        ecma_string_t *magic_string_p = ecma_get_magic_string (LIT_MAGIC_STRING_SOURCE);
+        ecma_property_t *prop_p = ecma_get_named_data_property (target_p, magic_string_p);
+        pattern_string_p = ecma_get_string_from_value (ecma_get_named_data_property_value (prop_p));
+        ecma_deref_ecma_string (magic_string_p);
+
+        /* Get flags. */
+        magic_string_p = ecma_get_magic_string (LIT_MAGIC_STRING_GLOBAL);
+        prop_p = ecma_get_named_data_property (target_p, magic_string_p);
+
+        if (ecma_is_value_true (ecma_get_named_data_property_value (prop_p)))
+        {
+          flags |= RE_FLAG_GLOBAL;
+        }
+
+        ecma_deref_ecma_string (magic_string_p);
+        magic_string_p = ecma_get_magic_string (LIT_MAGIC_STRING_IGNORECASE_UL);
+        prop_p = ecma_get_named_data_property (target_p, magic_string_p);
+
+        if (ecma_is_value_true (ecma_get_named_data_property_value (prop_p)))
+        {
+          flags |= RE_FLAG_IGNORE_CASE;
+        }
+
+        ecma_deref_ecma_string (magic_string_p);
+        magic_string_p = ecma_get_magic_string (LIT_MAGIC_STRING_MULTILINE);
+        prop_p = ecma_get_named_data_property (target_p, magic_string_p);
+
+        if (ecma_is_value_true (ecma_get_named_data_property_value (prop_p)))
+        {
+          flags |= RE_FLAG_MULTILINE;
+        }
+
+        ecma_deref_ecma_string (magic_string_p);
+
+        /* Get bytecode property. */
+        ecma_property_t *bc_prop_p = ecma_get_internal_property (this_obj_p,
+                                                                 ECMA_INTERNAL_PROPERTY_REGEXP_BYTECODE);
+        re_bytecode_t *old_bc_p = ECMA_GET_NON_NULL_POINTER (re_bytecode_t,
+                                                             bc_prop_p->u.internal_property.value);
+
+        FIXME ("We currently have to re-compile the bytecode, because we can't copy it without knowing its length.")
+        re_bytecode_t *new_bc_p = NULL;
+        ecma_completion_value_t bc_comp = re_compile_bytecode (&new_bc_p, pattern_string_p, flags);
+        /* Should always succeed, since we're compiling from a source that has been compiled previously. */
+        JERRY_ASSERT (ecma_is_completion_value_normal (bc_comp));
```

Comment:
`ecma_is_completion_value_empty`? In the case, completion value need not be freed.